### ISSUE TITLE
A logloss offset that rounds to 0 or 1 in Float32 stops training early

### ISSUE THREAD

**Issue by singhharsh1708** [Evovest/EvoTrees.jl#390]
An offset column of probabilities is converted to Float32 and then to logits, and `logit(1.0f0)` is `Inf`. Any Float64 probability within about 6e-8 of 0 or 1 lands on exactly `0.0f0` or `1.0f0` in that conversion, so an offset coming from a confident upstream model puts `±Inf` into the initial prediction. The iteration 0 metric is then `NaN`, `update_logger!` seeds `best_metric` with it, and every later comparison against `NaN` is false, so `iter_since_best` grows from the first round and training stops as soon as `early_stopping_rounds` is reached, even though the metric improves every round.

```julia
using EvoTrees, DataFrames, Random

rng = Xoshiro(8)
n = 1500
x = randn(rng, n, 3)
score = 3 .* x[:, 1] .+ x[:, 2]
y = Float64.(score .+ 0.3 .* randn(rng, n) .> 0)
off = 1 ./ (1 .+ exp.(-6 .* score))          # a confident upstream model, all in (0, 1)

df = DataFrame(x, :auto); df.y = y; df.off = off
cfg = EvoTreeRegressor(loss=:logloss, metric=:logloss, nrounds=200, max_depth=4,
                       eta=0.1, early_stopping_rounds=10, seed=1)
m = EvoTrees.fit(cfg, df; target_name="y", feature_names=["x1", "x2", "x3"],
                 offset_name="off", deval=df, verbosity=0)

println("offset rounds to exactly 0 or 1 for ",
        count(v -> v == 0f0 || v == 1f0, Float32.(off)), " of ", n, " rows")
println("rounds run  = ", length(m.trees))
println("best_iter   = ", m.info[:logger][:best_iter])
println("best_metric = ", m.info[:logger][:best_metric])
println("metrics     = ", round.(m.info[:logger][:metrics][1:6], digits=6))
```

```
offset rounds to exactly 0 or 1 for 276 of 1500 rows
rounds run  = 10
best_iter   = 0
best_metric = NaN
metrics     = [NaN, 0.060176, 0.058468, 0.05697, 0.055642, 0.054711]
```

The same data with the offset clipped to `[1e-6, 1-1e-6]` runs all 200 rounds and reaches 0.011958, so the model you get is about four times worse on the metric it was told to track. Nothing in the output says why.

`predict!` for `LogLoss` clamps every tree update to `[-15, 15]` (`src/predict.jl:25`) and the `MLE2P` and `MLogLoss` methods floor at -15, but the offset is added straight to `pred` in `init_core` (`src/init.jl:225`) and in both `CallBack` constructors (`src/callback.jl:100` and `:150`) with no clamp. The same missing clamp shows up under `:poisson`: a zero in the offset column gives `log(0.0)`, the `GradientRegression` `predict!` does not clamp either, and every logged metric comes out `Inf` rather than only the first.

This is the same shape as #354, where a metric that could never improve stopped training at `early_stopping_rounds`. There the target was outside the loss's support and rejecting it was the fix. Here the input is a valid probability and the `Inf` comes from EvoTrees' own Float32 conversion, so there is nothing to reject.

Would you rather the offset went through the same clamp `predict!` applies, or that `update_logger!` refused to seed `best_metric` from a non-finite iteration 0?
